### DISCUSSION TITLE
How to call action with middleware without schema

### DISCUSSION THREAD

**Discussion by aymericfontaine** [next-safe-action/next-safe-action#185] [Q&A] [answered]
Hi 👋

I am in the process of migrating a project from version 6 to version 7. I am encountering a compilation error when building my project related to an action without a schema.

Can you help me identify what I might have missed?

Thank you

```
const action = createSafeActionClient({
...
}).use(async ({ next }) => {
  const session = await auth();
  return next({ ctx: { user: session?.user } });
});

export const authAction = action.use(async ({ next, ctx }) => {
  if (!ctx.user) throw new ActionUnauthorizedError();
  return next({ ctx: { user: ctx.user } });
});
```

```
// no schema specified
export const getWallets = authAction.action(async ({ ctx: { user } }) => {
 return await prisma.wallet.findMany({
    where: {
      userId: user.id,
    }
  });
});
```

```
export default async function Page() {
  const wallets = await getWallets();

  if (!wallets?.data) {
    redirect("/");
  }

  ...
```

```
Type error: Expected 1 arguments, but got 0.
Expected first argument of getWallets() is "input: void"
```

**Comment by TheEdoRan** [comment]
Well, that's pretty weird. Type `void` means that you don't need to provide an argument, so it's correct. What's even weirder is that I've just tried to compile a Next.js project with your same action setup and got no compilation errors:

![image](https://github.com/TheEdoRan/next-safe-action/assets/1337629/740ba5ce-99dc-4330-b250-4751ed5e43eb)

Calling it in a client component page with no arguments:

![image](https://github.com/TheEdoRan/next-safe-action/assets/1337629/610475ec-cbba-4d77-8cb1-8faabbf63cd2)

And here's the action client file:
 
![image](https://github.com/TheEdoRan/next-safe-action/assets/1337629/153a4ba8-1249-4afc-8e32-0c5d3a647b7f)

**Reply by aymericfontaine** [reply]
Thank you for the feedback. It is indeed strange. My VSCode does not show any errors, but it does not compile during the build. I have the same problem when deploying on Vercel.

The solution I found is to send undefined as a parameter.

<img width="720" alt="image" src="https://github.com/TheEdoRan/next-safe-action/assets/39033239/e0f232f1-7e06-4223-954c-2b39b34e5ad3">

**Reply by TheEdoRan** [reply]
If you can reproduce it in a new repo/project, please provide a link to it so I can investigate this issue, because that's very weird. Never seen that happen before. Thanks!

**Comment by aymericfontaine** [comment, marked as answer]
Hey @TheEdoRan, my issue was due to a TypeScript version >= 5.5.2
The latest experimental version (https://github.com/TheEdoRan/next-safe-action/issues/180#issuecomment-2201607407) fixes the problem 👍
